src/server/cluster/cluster_config.h

```diff
@@ -36,10 +36,18 @@ class ClusterConfig {
     uint16_t port = 0;
   };
 
+  struct MigrationInfo {
+    std::vector<SlotRange> slot_ranges;
+    std::string target_id;
+    std::string ip;
+    uint16_t port = 0;
+  };
+
   struct ClusterShard {
     SlotRanges slot_ranges;
     Node master;
     std::vector<Node> replicas;
+    std::vector<MigrationInfo> migrations;
```

Comment:
I think this can be an optional field

Reply:
I don't see any point to make it optional if it can be empty

Reply:
all the places that you write .migrations = {} can be removed than?
this is just optional data and I think setting .migrations = {} is redundant

Reply:
It's a compilation warning, because not members are initialized

Reply:
and if we change it to optional?